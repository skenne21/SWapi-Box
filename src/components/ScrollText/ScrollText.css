.ScrollText {
  margin: auto;
  background-image: url('../../assets/text-background.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  width: 60%;
  height: 100%;
  overflow: hidden;
}

.main-container__text {
  display: flex;
  justify-content: center;
  height: 400px;
  perspective: 400px;
  font-size: 500%;
  line-height: 150%;
  letter-spacing: 6px;
  color: yellow;
  text-align: justify;
  margin: auto;
}

.moving-text-box {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: moveing-text-box 60s linear;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

@keyFrames moveing-text-box {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
   100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
   }
}