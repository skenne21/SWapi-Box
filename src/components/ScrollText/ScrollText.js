import React from 'react';
import PropTypes from 'prop-types';
import './ScrollText.css'

const ScrollText = ({film}) => {
  const { title, opening, release} = film;
  return (
    <section className="ScrollText">
      <div className='fade'> </div>
      <div className="main-container__text">
        <div className="moving-text-box">
          <header className="title">
            <p>{title}</p>
            <h1>{release}</h1>
          </header>
          <p className='opening-text'>{opening}</p>
        </div>
      </div>
    </section>

  )
}

export default ScrollText;

ScrollText.propTypes = {
  film:PropTypes.object.isRequired
}
