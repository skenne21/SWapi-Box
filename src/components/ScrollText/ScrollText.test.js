import React from 'react';
import { shallow } from 'enzyme';
import ScrollText from './ScrollText.js';

describe('ScrollText', () => {
  it('Should match the snapshot', () => {
    const film = {
      title: 'Attack of the clones',
      opening: 'It is a dark time for...',
      year: '2002'
    };
    const wrapper = shallow(<ScrollText film={film} />);
    expect(wrapper).toMatchSnapshot();
  });
});