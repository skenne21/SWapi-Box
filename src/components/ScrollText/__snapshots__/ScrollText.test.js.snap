// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollText
    film={
        Object {
            "opening": "It is a dark time for...",
            "title": "Attack of the clones",
            "year": "2002",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
>
           
</div>,
        <div
          className="main-container__text"
>
          <div
                    className="moving-text-box"
          >
                    <header
                              className="title"
                    >
                              <p>
                                        Attack of the clones
                              </p>
                              <h1 />
                    </header>
                    <p
                              className="opening-text"
                    >
                              It is a dark time for...
                    </p>
          </div>
</div>,
      ],
      "className": "ScrollText",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ",
          "className": "fade",
        },
        "ref": null,
        "rendered": " ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="moving-text-box"
>
            <header
                        className="title"
            >
                        <p>
                                    Attack of the clones
                        </p>
                        <h1 />
            </header>
            <p
                        className="opening-text"
            >
                        It is a dark time for...
            </p>
</div>,
          "className": "main-container__text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header
                className="title"
>
                <p>
                                Attack of the clones
                </p>
                <h1 />
</header>,
              <p
                className="opening-text"
>
                It is a dark time for...
</p>,
            ],
            "className": "moving-text-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Attack of the clones
</p>,
                  <h1 />,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Attack of the clones",
                  },
                  "ref": null,
                  "rendered": "Attack of the clones",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
              ],
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It is a dark time for...",
                "className": "opening-text",
              },
              "ref": null,
              "rendered": "It is a dark time for...",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
>
             
</div>,
          <div
            className="main-container__text"
>
            <div
                        className="moving-text-box"
            >
                        <header
                                    className="title"
                        >
                                    <p>
                                                Attack of the clones
                                    </p>
                                    <h1 />
                        </header>
                        <p
                                    className="opening-text"
                        >
                                    It is a dark time for...
                        </p>
            </div>
</div>,
        ],
        "className": "ScrollText",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
            "className": "fade",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="moving-text-box"
>
              <header
                            className="title"
              >
                            <p>
                                          Attack of the clones
                            </p>
                            <h1 />
              </header>
              <p
                            className="opening-text"
              >
                            It is a dark time for...
              </p>
</div>,
            "className": "main-container__text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <header
                  className="title"
>
                  <p>
                                    Attack of the clones
                  </p>
                  <h1 />
</header>,
                <p
                  className="opening-text"
>
                  It is a dark time for...
</p>,
              ],
              "className": "moving-text-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Attack of the clones
</p>,
                    <h1 />,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Attack of the clones",
                    },
                    "ref": null,
                    "rendered": "Attack of the clones",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h1",
                  },
                ],
                "type": "header",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "It is a dark time for...",
                  "className": "opening-text",
                },
                "ref": null,
                "rendered": "It is a dark time for...",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
